rest/replicator_test.go

```diff
@@ -2530,6 +2531,212 @@ func TestActiveReplicatorIgnoreNoConflicts(t *testing.T) {
 	assert.Equal(t, strconv.FormatUint(localDoc.Sequence, 10), ar.GetStatus().LastSeqPush)
 }
 
+// TestActiveReplicatorReconnectOnStart ensures ActiveReplicators retry their initial connection for cases like:
```

Comment:
I think it would be useful to have tests for the following scenarios, to validate the current behaviour and prevent regressions:
1. Issuing _replicationStatus?action=stop for a replication that's in the reconnect retry loop
2. Issuing _replicationStatus?action=start for a replication that's in the reconnect retry loop
3. Trigger rebalance for a replication that's in the reconnect retry loop

Reply:
Have a lower level test (`TestActiveReplicatorReconnectSendActions`) which calls `Stop()` on a replicator in the middle of the reconnect loop to cover these cases.